src/Symfony/Component/Messenger/Transport/Semaphore/Connection.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Semaphore;
+
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Component\Messenger\Transport\Semaphore\Exception\SemaphoreException;
+
+/**
+ * A Semaphore connection.
+ *
+ * @author Cedrick Oka Baidai <okacedrick@gmail.com>
+ */
+class Connection
+{
+    private const DEFAULT_OPTIONS = [
+        'project' => 'M',
+        'message_max_size' => 131072,
+        'auto_setup' => true,
+    ];
+
+    private $configuration;
+
+    /**
+     * @var resource
+     */
+    private $queue;
+
+    /**
+     * @var bool
+     */
+    private $setup;
+
+    public function __construct(array $configuration)
+    {
+        if (false === \extension_loaded('sysvmsg')) {
+            throw new TransportException('Semaphore extension (sysvmsg) is required.');
+        }
+
+        $this->configuration = array_replace_recursive(self::DEFAULT_OPTIONS, $configuration);
+        $this->setup = false;
+
+        $this->configuration['message_max_size'] = (int) $this->configuration['message_max_size'];
+
+        if (true === isset($this->configuration['message_type'])) {
+            $this->configuration['message_type'] = (int) $this->configuration['message_type'];
+        }
+    }
+
+    /**
+     * Creates a connection based on the DSN and options.
+     *
+     * Available options:
+     *
+     *   * path: Pathname to create System V IPC key (Default: __FILE__)
+     *   * project: Project identifier to create System V IPC key (Default: M)
+     *   * message_type: The type of message to send (Default: 1)
```

Comment:
does those 2 parameter have to be customisalbe by the user? 
Isn't a unique path enougth to use this transport?